### PR TITLE
raise exception when an empty config file is passed to load_kube_config

### PR DESCRIPTION
**Pull request by MridulS**
`load_kube_config` currently fails with a nondescript `TypeError: 'NoneType'` error if an empty kube_config file is used.

Error log
```
~/miniconda3/envs/dask/lib/python3.8/site-packages/kubernetes/config/kube_config.py in load_kube_config(config_file, context, client_configuration, persist_config)
    790         config_file = KUBE_CONFIG_DEFAULT_LOCATION
    791
--> 792     loader = _get_kube_config_loader(
    793         filename=config_file, active_context=context,
    794         persist_config=persist_config)

~/miniconda3/envs/dask/lib/python3.8/site-packages/kubernetes/config/kube_config.py in _get_kube_config_loader(filename, config_dict, persist_config, **kwargs)
    744         **kwargs):
    745     if config_dict is None:
--> 746         kcfg = KubeConfigMerger(filename)
    747         if persist_config and 'config_persister' not in kwargs:
    748             kwargs['config_persister'] = kcfg.save_changes

~/miniconda3/envs/dask/lib/python3.8/site-packages/kubernetes/config/kube_config.py in __init__(self, paths)
    671             self._load_config_from_file_like_object(paths)
    672         else:
--> 673             self._load_config_from_file_path(paths)
    674
    675     @property

~/miniconda3/envs/dask/lib/python3.8/site-packages/kubernetes/config/kube_config.py in _load_config_from_file_path(self, string)
    693                 if os.path.exists(path):
    694                     self.paths.append(path)
--> 695                     self.load_config(path)
    696         self.config_saved = copy.deepcopy(self.config_files)
    697

~/miniconda3/envs/dask/lib/python3.8/site-packages/kubernetes/config/kube_config.py in load_config(self, path)
    703             config_merged = copy.deepcopy(config)
    704             for item in ('clusters', 'contexts', 'users'):
--> 705                 config_merged[item] = []
    706             self.config_merged = ConfigNode(path, config_merged, path)
    707         for item in ('clusters', 'contexts', 'users'):

TypeError: 'NoneType' object does not support item assignment
```

I have added a check to raise a `ConfigException` error if an empty file is encountered.

### CODE REVIEW

**Comment by MridulS** [comment]
Signed the CLA.

**Comment by MridulS** [comment]
The travis error on functional tests seems to be because of a minikube error.

```
W0125 15:54:45.089355    4821 out.go:181] X Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 1 is less than the minimum allowed of 2
493X Exiting due to RSRC_INSUFFICIENT_CORES: Requested cpu count 1 is less than the minimum allowed of 2
494I0125 15:54:45.089376    4821 out.go:119] 
495
496
```

File: config/kube_config_test.py [hunk 1]
```diff
@@ -1290,6 +1290,21 @@ def test_load_kube_config_from_dict(self):
                                    client_configuration=actual)
         self.assertEqual(expected, actual)
 
+    def test_load_kube_config_from_empty_file(self):
```
Comment (fabianvf) [inline config/kube_config_test.py:1293]: it looks like these test names are flipped

Comment (MridulS) [reply config/kube_config_test.py]: whoops! indeed thanks for the catch!

**Comment by MridulS** [comment]
Travis seems to be failing due to an unrelated test. Any ideas what's happening here?

**Comment by MridulS** [comment]
Just wanted to check in to see if this looks okay :)

**Review by fabianvf — APPROVED** [review]

**Pull request merged**